fuel-core/src/schema/balance.rs

```diff
@@ -118,6 +120,10 @@ impl BalanceQuery {
         before: Option<String>,
     ) -> async_graphql::Result<Connection<AssetId, Balance, EmptyFields, EmptyFields>>
     {
+        if check_pagination_parameters(&first, &after, &last, &before) {
```

Comment:
We have this check "(first.is_some() && before.is_some())" in 7 places, but you only updated it in 4
<img width="299" alt="image" src="https://user-images.githubusercontent.com/18346821/203814847-c487a723-9492-4ecc-9d80-f1771d2e64ca.png">

Maybe better to create a new function, `query` or `query_local`, and use it everywhere with this check inside=)